wallet/src/main/scala/org/bitcoins/wallet/api/WalletApi.scala

```diff
@@ -99,12 +104,14 @@ trait LockedWalletApi extends WalletApi with WalletLogger {
       scriptPubKeys = utxos.flatMap(_.redeemScriptOpt).toSet ++ addresses
         .map(_.scriptPubKey)
         .toSet
-      _ <- Future {
-        val matcher = SimpleFilterMatcher(blockFilter)
-        if (matcher.matchesAny(scriptPubKeys.toVector.map(_.asmBytes))) {
-          nodeApi.downloadBlocks(Vector(blockHash))
-        }
+      processFs = blockFilters.map {
+        case (blockHash, blockFilter) =>
+          val matcher = SimpleFilterMatcher(blockFilter)
+          if (matcher.matchesAny(scriptPubKeys.toVector.map(_.asmBytes))) {
+            nodeApi.downloadBlocks(Vector(blockHash))
+          } else FutureUtil.unit
       }
+      _ <- Future.sequence(processFs)
```

Comment:
Do we care about the block order here? `Future.sequence` doesn't guarantee the order.

Also, what is error recovery strategy here?